### PR TITLE
[ETFE-2372] Added DispatchUseConsignorDetails page

### PR DESCRIPTION
**Pull request by MJCallahanPage**

### CODE REVIEW

File: conf/dispatch.routes [hunk 1]
```diff
@@ -13,3 +13,9 @@ POST        /trader/:ern/draft/:draftId/dispatch/address/change              con
...
 POST        /trader/:ern/draft/:draftId/dispatch/check-answers               controllers.sections.dispatch.DispatchCheckAnswersController.onSubmit(ern: String, draftId: String)
+
+# DIS02
+GET        /trader/:ern/draft/:draftId/dispatch/consignor-details            controllers.sections.dispatch.DispatchUseConsignorDetailsController.onPageLoad(ern: String, draftId: String, mode: Mode = NormalMode)
+POST       /trader/:ern/draft/:draftId/dispatch/consignor-details            controllers.sections.dispatch.DispatchUseConsignorDetailsController.onSubmit(ern: String, draftId: String, mode: Mode = NormalMode)
+GET        /trader/:ern/draft/:draftId/Dispatch/consignor-details/change     controllers.sections.dispatch.DispatchUseConsignorDetailsController.onPageLoad(ern: String, draftId: String, mode: Mode = CheckMode)
+POST       /trader/:ern/draft/:draftId/Dispatch/consignor-details/change     controllers.sections.dispatch.DispatchUseConsignorDetailsController.onSubmit(ern: String, draftId: String, mode: Mode = CheckMode)
```
Comment (LewisMitchell) [inline conf/dispatch.routes:21]:
```suggestion
GET        /trader/:ern/draft/:draftId/dispatch/consignor-details/change     controllers.sections.dispatch.DispatchUseConsignorDetailsController.onPageLoad(ern: String, draftId: String, mode: Mode = CheckMode)
POST       /trader/:ern/draft/:draftId/dispatch/consignor-details/change     controllers.sections.dispatch.DispatchUseConsignorDetailsController.onSubmit(ern: String, draftId: String, mode: Mode = CheckMode)
```

File: app/viewmodels/checkAnswers/sections/dispatch/DispatchUseConsignorDetailsSummary.scala [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+        val value = if (answer) "site.yes" else "site.no"
+
+        SummaryListRowViewModel(
+          key     = "dispatchUseConsignorDetails.checkYourAnswersLabel",
+          value   = ValueViewModel(value),
+          actions = Seq(
+            ActionItemViewModel(
+              "site.change",
+              controllers.sections.dispatch.routes.DispatchUseConsignorDetailsController.onPageLoad(request.ern, request.draftId, CheckMode).url,
+              "changeDispatchUseConsignorDetails"
```
Comment (LewisMitchell) [inline app/viewmodels/checkAnswers/sections/dispatch/DispatchUseConsignorDetailsSummary.scala:42]: Is it worth naming the parameters?

File: app/pages/sections/dispatch/DispatchUseConsignorDetailsPage.scala [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+package pages.sections.dispatch
+
+import pages.QuestionPage
+import play.api.libs.json.JsPath
+
+case object DispatchUseConsignorDetailsPage extends QuestionPage[Boolean] {
+
+  override val path: JsPath = JsPath \ "dispatch" \ toString
+
+  override def toString: String = "dispatchUseConsignorDetails"
```
Comment (LewisMitchell) [inline app/pages/sections/dispatch/DispatchUseConsignorDetailsPage.scala:26]:
```suggestion
  override val toString: String = "dispatchUseConsignorDetails"
  override val path: JsPath = DispatchSection.path \ toString
```

File: app/controllers/sections/dispatch/DispatchUseConsignorDetailsController.scala [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+  def onSubmit(ern: String, draftId: String, mode: Mode): Action[AnyContent] =
+    authorisedDataRequestAsync(ern, draftId) { implicit request =>
+      formProvider().bindFromRequest().fold(
+        renderView(BadRequest, _, mode),
+        saveAndRedirect(DispatchUseConsignorDetailsPage, _, mode)
+      )
+    }
+
+  private def renderView(status: Status, form: Form[Boolean], mode: Mode)(implicit request: DataRequest[_]): Future[Result] =
+    Future.successful(status(view(form, routes.DispatchUseConsignorDetailsController.onSubmit(request.ern, request.draftId, mode))))
```
Comment (LewisMitchell) [inline app/controllers/sections/dispatch/DispatchUseConsignorDetailsController.scala:61]: I'm not sure why we need to pass the submit call through, it's pretty static so should be in the view IMO. What do you think?

**Review by LewisMitchell — CHANGES_REQUESTED** [review]
The CYA page helper needs updating to include the row you added: https://github.com/hmrc/emcs-tfe-create-movement-frontend/blob/8dfa947fb444eb171a2242097c0d594390ecc0bb/app/viewmodels/checkAnswers/sections/dispatch/DispatchCheckAnswersHelper.scala#L28

**Comment by LewisMitchell** [comment]
Could you rebase as well, please?

File: app/controllers/sections/dispatch/DispatchUseConsignorDetailsController.scala (hunk 4 above)
Comment (james-work-account) [reply app/controllers/sections/dispatch/DispatchUseConsignorDetailsController.scala:69]: Maybe one for a wider refactoring task down the line :)

**Review by r-melvin — APPROVED** [review]

**Pull request merged**